File: src/main/java/com/metradingplat/marketdata/domain/models/Candle.java
package com.metradingplat.marketdata.domain.models;

import java.time.Instant;
import java.time.LocalDateTime;
import java.time.ZoneId;

import com.metradingplat.marketdata.domain.enums.EnumTimeframe;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Candle {
    private String symbol;
    private EnumTimeframe timeframe;
    private Instant timestamp;
    private Double open;
    private Double high;
    private Double low;
    private Double close;
    private Double volume;

    public LocalDateTime getLocalDateTime() {
        return LocalDateTime.ofInstant(timestamp, ZoneId.systemDefault());
    }
}

File: internal/core/domain/candle.go
package domain

import "time"

// Tags json en el formato exacto que ya esperaba el frontend del servicio
// Java (HistoricalCandleDTO: symbol/timestamp/open/high/low/close/volume/vwap).
type Candle struct {
	Symbol     string    `json:"symbol"`
	Timeframe  Timeframe `json:"timeframe"`
	Timestamp  time.Time `json:"timestamp"`
	Open       float64   `json:"open"`
	High       float64   `json:"high"`
	Low        float64   `json:"low"`
	Close      float64   `json:"close"`
	Volume     int64     `json:"volume"`
	TradeCount int       `json:"tradeCount"`
	VWAP       float64   `json:"vwap"`
	Source     string    `json:"source"`
}

func (c Candle) IsComplete() bool {
	return !c.Timestamp.IsZero() && c.Open != 0 && c.High != 0 && c.Low != 0 && c.Close != 0
}
